OlinkAnalyze/tests/testthat/test-Read_NPX_data.R

```diff
@@ -1,142 +1,156 @@
-    "Flex data in long form detected"
-  )
-
-  expect_equal(input$Quantified_value, c(0.19169, 336.12903))
-})
-
-# No warning for extra column ----------------
-
-test_that("extra column", {
-  expect_no_warning(read_NPX(testthat::test_path("refs/mock_sampleID_hashes.csv")))
-})
+test_that("Data loads correctly with 'read_NPX()'", {
+  #Read data
+  npx_file <- system.file("extdata", "npx_data1.xlsx", package = "OlinkAnalyze", mustWork = TRUE)
+  manifest_file <- system.file("extdata", "npx_data1_meta.csv", package = "OlinkAnalyze", mustWork = TRUE)
+  parquet_file <- system.file("extdata", "npx_data_ext.parquet",
+                              package = "OlinkAnalyze", mustWork = TRUE)
+  df_1 <- read_NPX(filename = npx_file) # load dataset 1
+  manifest_1 <- read.delim(manifest_file, header = TRUE, sep = ';')
+
+  # load manifest 1
+  df_2_2 <- read_NPX(system.file("extdata", "Example_NPX_Data2_1.csv", package = "OlinkAnalyze", mustWork = TRUE))
+  npx_file_v3 <- system.file("extdata", "npx_data_v3.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  df_v3 <- read_NPX(filename = npx_file_v3)
+  npx_file_ext_v1 <- system.file("extdata", "npx_data_ext_v1.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  df_ext_v1 <- read_NPX(filename = npx_file_ext_v1)
+  npx_file_ext_v2 <- system.file("extdata", "npx_data_ext_v2.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  df_ext_v2 <- read_NPX(filename = npx_file_ext_v2)
+  df_parquet <- read_NPX(filename = parquet_file)
+
+  #NPX read ok?
+  expect(exists("df_1"), failure_message = "read_NPX failed on dataset 1")
+  expect_s3_class(df_1, class = "tbl_df")
+  expect(exists("df_v3"), failure_message = "read_NPX failed on dataset v3")
+  expect(exists("df_ext_v1"), failure_message = "read_NPX failed on extended dataset v1")
+  expect(exists("df_ext_v2"), failure_message = "read_NPX failed on extended dataset v2")
+  expect(exists("df_parquet"), failure_message = "read_NPX failed on parquet dataset")
+
+  #NPX zip read ok?
+  zip_npx_file_fail_1 <- system.file("extdata", "Example_NPX_Data_zip.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  expect_error(read_NPX(filename = zip_npx_file_fail_1), "Checksum of NPX file does not match the one from \"MD5_checksum.txt\"! Loss of data?")
+
+  zip_npx_file_fail_2 <- system.file("extdata", "Example_NPX_Data_empty.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  expect_error(read_NPX(filename = zip_npx_file_fail_2), "The compressed file does not contain a valid NPX file. Expecting: \"README.txt\", \"MD5_checksum.txt\" or \"checksum_sha256.txt\" and the NPX file.")
+
+  zip_npx_file_success <- system.file("extdata", "Example_NPX_Data_3K.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  df_2 <- read_NPX(filename = zip_npx_file_success)
+
+  zip_npx_file_success_sha <- system.file("extdata", "Example_NPX_Data_sha256.zip", package = "OlinkAnalyze", mustWork = TRUE)
+  expect_snapshot(read_NPX(filename = zip_npx_file_success_sha))
```

Comment:
expect_snapshot(read_NPX(filename = zip_npx_file_success_sha))

throws a fail for me. Not sure really what this is suppose to do. It compares to a previous run of it?

Reply:
@Orbmac Interesting, it doesnt fail locally for me.  
expect_snapshot compares the output of reading the file to the Read_NPX_data.md in tests/testthat/_snaps. 

If you run 
`zip_npx_file_success_sha <- system.file("extdata", "Example_NPX_Data_sha256.zip", package = "OlinkAnalyze", mustWork = TRUE)`
and then 
`read_NPX(filename = zip_npx_file_success_sha)`

are there any differences in the console output compared to the Read_NPX_data.md file?

Reply:
I assume its something locally on my computer then if it works for you. The values was the same, the only difference I could see was that it said "Missing~1" while for me it said ""Missing..1" (similar for other long col names).